python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -0,0 +1,682 @@
+                                calibration_label=self.calibrationLabel,
+                                producer=None)
+
+    def convertStorageClass(self, data, datasetTypeName):
+        """Switch from an exposure to the image type expected.
+
+        Parameters
+        ----------
+        data : `lsst.afw.image.Exposure`
+            Input exposure data to convert.
+        datasetTypeName : `str`
+            Dataset type that will be registered
+
+        Returns
+        -------
+        data : `lsst.afw.image.Image` or `lsst.afw.image.MaskedImage`
+            Converted image data to register.
+        """
+        if datasetTypeName in ('bias', 'dark'):
+            data = data.getImage()
+        elif datasetTypeName in ('flat', ):
+            data = data.getMaskedImage()
+        else:
+            pass
+        return data
+
+    def registerDatasetType(self, data, datasetTypeName, dataId):
+        storageClass = 'ExposureF'
+        if datasetTypeName in ('bias', 'dark'):
+            storageClass = 'ImageF'
+        elif datasetTypeName in ('flat', ):
+            storageClass = 'MaskedImageF'
+
+        dimensionArray = list(dataId.keys()) + ["calibration_label"]
+        datasetType = DatasetType(datasetTypeName,
+                                  set(dimensionArray),
+                                  storageClass,
+                                  universe=self.butler.registry.dimensions)
+        self.butler.registry.registerDatasetType(datasetType)
+
+    def addCalibrationLabel(self, name=None, instrument=None,
+                            beginDate="1970-01-01", endDate="2038-12-31"):
+        """Method to allow tasks to add calibration_label for master calibrations.
+
+        Parameters
+        ----------
+        name : `str`
+            A unique string for the calibration_label key.
+        instrument : `str`
+            Instrument this calibration is for.
+        beginDate : `str`
+            An ISO 8601 date string for the beginning of the valid date range.
+        endDate : `str`
+            An ISO 8601 date string for the end of the valid date range.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if the instrument or calibration_label name are not set.
+        """
+        if name is None:
+            name = self.calibrationLabel
+        if instrument is None:
+            instrument = self.instrument
+        if name is None or instrument is None:
+            raise RuntimeError("Instrument and calibration_label name not set.")
+
+        try:
+            existingValues = self.registry.queryDimensions(['calibration_label'],
+                                                           instrument=self.instrument,
+                                                           calibration_label=name)
+            existingValues = [a for a in existingValues]
+            print(f"Found {len(existingValues)} Entries for {self.calibrationLabel}")
```

Comment:
You need to tell the user here that this means you're not going to do anything. In particular, they may be expecting to change the `beginDate` and `endDate` (how do I do that if this won't let me do it?).
Why use `print` instead of `self.log`?